src/helpers/MiscFunctions.cpp

```diff
@@ -330,8 +331,28 @@ int getWorkspaceIDFromString(const std::string& in, std::string& outName) {
 
             std::sort(validWSes.begin(), validWSes.end());
 
+            if (includeEmpty && validWSes.size() >= 2) {
+                // Include all empty workspaces in between the known workspaces
+                // We cannot just add all empty workspaces beforehand, since then validWSes would be potentially of size 2^31-1
+                for (auto it = validWSes.begin(); it != validWSes.end() - 1; ++it) {
+                    // add all empty workspaces in between current and last
+                    int curValidWSID  = *it;
+                    int nextValidWSID = *(it + 1);
+                    // ]curValidWSID, nextValidWSID[
+                    for (int id = curValidWSID + 1; id < nextValidWSID; id++) {
+                        const auto PWORKSPACE = g_pCompositor->getWorkspaceByID(id);
+                        if (!PWORKSPACE || (g_pCompositor->getWindowsOnWorkspace(id) == 0 && PWORKSPACE->m_iMonitorID == g_pCompositor->m_pLastMonitor->ID)) {
```

Comment:
not checking for whether it's bound to a different monitor

Reply:
Can a nullptr workspace be bound on another monitor? Otherwise its already being checked.

Reply:
yes, a workspace can be closed but bound. Check CConfigManager::getBoundMonitorForWS

Reply:
Thanks for tip, I just copied what was in l.325 (I guess that's faulty too...)